src/coreclr/nativeaot/Runtime/MiscHelpers.cpp

```diff
@@ -356,12 +356,17 @@ EXTERN_C NATIVEAOT_API int32_t __cdecl RhpGetCurrentThreadStackTrace(void* pOutp
     return RhpCalculateStackTraceWorker(pOutputBuffer, outputBufferLength, pAddressInCurrentFrame);
 }
 
-COOP_PINVOKE_HELPER(void*, RhpRegisterFrozenSegment, (void* pSegmentStart, size_t length))
+COOP_PINVOKE_HELPER(void*, RhRegisterFrozenSegment, (void * pSection, size_t allocSize, size_t commitSize, size_t reservedSize))
```

Comment:
`RhRegisterFrozenSegment` is a regular PInvoke called in preeemptive mode. It should not be defined using `COOP_PINVOKE_HELPER` macro.

Reply:
(same for other frozen segment methods)